docs/blocks/custom-fluids.md

```diff
@@ -0,0 +1,851 @@
+            }
+          },
+
+          {
+            "condition": "!query.block_neighbor_has_any_tag(0, 0, -1, 'template_full')",
+            "set_block_property": {
+              "fluids:nz": 0
+            }
+          },
+          {
+            "condition": "!query.block_neighbor_has_any_tag(0, 0, 1, 'template_full')",
+            "set_block_property": {
+              "fluids:z": 0
+            }
+          },
+          {
+            "condition": "!query.block_neighbor_has_any_tag(1, 0, 0, 'template_full')",
+            "set_block_property": {
+              "fluids:x": 0
+            }
+          },
+          {
+            "condition": "!query.block_neighbor_has_any_tag(-1, 0, 0, 'template_full')",
+            "set_block_property": {
+              "fluids:nx": 0
+            }
+          },
+          {
+            "condition": "!query.block_neighbor_has_any_tag(0, 1, 0, 'template_full')",
+            "set_block_property": {
+              "fluids:top": 0
+            }
+          },
+          {
+            "condition": "!query.block_neighbor_has_any_tag(0, -1, 0, 'template_full')",
+            "set_block_property": {
+              "fluids:bottom": 0
+            }
+          },
+
+          {
+            "condition": "query.block_neighbor_has_any_tag(0, 1, 0, 'template')",
+            "run_command": {
+              "target": "self",
+              "command": [ "execute unless block ~ ~-1 ~ air unless block ~ ~-1 ~ fluids:fluid_template unless block ~ ~-1 ~ fluids:fluid_template1 unless block ~ ~-1 ~ fluids:fluid_template2 unless block ~ ~-1 ~ fluids:fluid_template3 unless block ~ ~-1 ~ fluids:fluid_template_down run setblock ~ ~ ~ fluids:fluid_template_down" ]
+            }
+          }
+        ]
+      }
+    }
+  }
+}
+```
+
+</Spoiler>
+
+## Fluid Bucket
+
+To pickup or place your custom fluid you need a custom bucket item. Although any item can pickup the fluid, your fluid can be customized to require this custom bucket. Below is the JSON for the custom bucket. To replicate the json, copy and quick replace `template` with your fluid's name.
+
+<CodeHeader>BH\items\template_bucket.json</CodeHeader>
+<Spoiler title="Template Bucket JSON">
+
+```json
+{
+  "format_version": "1.16.100",
+  "minecraft:item": {
+    "description": {
+      "identifier": "fluids:template_bucket",
+      "category": "items"
+    },
+    "components": {
+      "minecraft:icon": {
+        "texture": "template_bucket"
+      },
+      "minecraft:display_name": {
+        "value": "Template Bucket"
+      },
+      "minecraft:max_stack_size": 1,
+      "minecraft:block_placer": {
+        "block": "fluids:fluid_template"
+      }
+    }
+  }
+}
+```
+
+</Spoiler>
+
+## Scripts
+
+The fluids use a script to add the ability for the player to float/sink in the fluid. The script also adds fog. To add your fluid to the script, put the ID of your new fluids in the `fluidsIDs` string array.
+
+<CodeHeader>BH\scripts\fluids\main.js</CodeHeader>
```

Comment:
Please make the codeheader follow the same style guide, `BP/scripts/filename.js`